### PR TITLE
tests: fix failing tests on NixOS

### PR DESCRIPTION
**Pull request by mingmingrr**
In test/backend/x11/test_window.py: `subprocess.run(["import", ...])` fails with:
`FileNotFoundError: [Errno 2] No such file or directory: 'import'`
For some reason, this errors even if imagemagick is installed hence the `shutil.which`.

In libqtile/scripts/migrations/match_list_regex.py, a missing `isort` import causes test_all_migrations[MatchListRegex-0] to fail. Adding it to common-python-deps fixes this. Related issue: #4658

### CODE REVIEW

**Comment by tych0** [comment]
> For some reason, this errors even if imagemagick is installed hence the shutil.which.

Hmm, that seems odd. Can you post the `$PATH` when `shutil.which` works, and where the binary is finally located when it does?

> In libqtile/scripts/migrations/match_list_regex.py, a missing isort import causes test_all_migrations[MatchListRegex-0] to fail. Adding it to common-python-deps fixes this.

this part looks good to me; you could separate it out as a separate patch and I'd be happy to land that.

**Comment by mingmingrr** [comment]
Here's my `$PATH`: [path.txt](https://github.com/user-attachments/files/17970641/path.txt)
`print(shutil.which("import"))` shows it's at `/nix/store/irkcg78qfr4q5hfmjw8k6w92f8bgb39f-imagemagick-7.1.1-38/bin/import`

I did some more digging with a modified subprocess module. Turns out when `env=` is set in `subprocess.run()`, it gets passed along to `os.get_exec_path`, and the current `$PATH` is not automatically inserted, in which case it uses a fallback of `PATH=/usr/bin:/bin`. So either `$PATH` would need to be added to `env=`, or a filepath would need to be passed as the executable.

TLDR: to make that test work on NixOS, we would either need `shutil.which`, or add `$PATH` to `subprocess.run(env=...)`

**Comment by tych0** [comment]
Nice, that makes sense. Can you add that second blurb to your commit message? Then I think it looks good to me!

**Comment by tych0** [comment]
Arg. I tried to update with the commit log, and it force closed the PR and I can't reopen it.

**Comment by tych0** [comment]
idk, I reopened as #5113; I will land it from there. Thanks for the patch!
